pyat/at/lattice/axisdef.py

```diff
@@ -0,0 +1,163 @@
+_axis_def[Ellipsis] = dict(index=Ellipsis, label="", unit="", code="...")
+
+_plane_def = dict(
+    h=dict(index=0, label="h", unit=" [m]"),
+    v=dict(index=1, label="v", unit=" [m]"),
+    l=dict(index=2, label="l", unit="")
+)
+for xk, xv in [it for it in _plane_def.items()]:
+    xv['code'] = xk
+    _plane_def[xv['index']] = xv
+    _plane_def[xk.upper()] = xv
+_plane_def['x'] = _plane_def['h']
+_plane_def['y'] = _plane_def['v']
+_plane_def[None] = dict(index=slice(None), label="", unit="", code=":")
+_plane_def[Ellipsis] = dict(index=Ellipsis, label="", unit="", code="...")
+
+
+def _descr(dd: dict, arg: AxisDef, key: Optional[str] = None):
+    if isinstance(arg, tuple):
+        return tuple(_descr(dd, a, key=key) for a in arg)
+    else:
+        try:
+            descr = dd[arg]
+        except (TypeError, KeyError):
+            descr = dict(index=arg, code=arg, label="", unit="")
+        if key is None:
+            return descr
+        else:
+            return descr[key]
+
+
+def axis_(axis: AxisDef, key: Optional[str] = None):
+    r"""Return axis descriptions
+
+    Parameters:
+        axis:           axis code or tuple of axis codes selecting axes in the
+          standard AT coordinate system. codes for the 6 axes are:
+
+          0, 'x', 'X'
+
+          1, 'px', PX', 'xp'
+
+          2, 'z', 'Z'
+
+          3, 'py', 'PY', 'yp'
+
+          4, 'dp', 'delta'
+
+          5, 'ct'
+
+          :py:obj:`None`, slice(None) and :py:obj:`Ellipsis` selects all axes
+
+        key:            key in the coordinate descriptor dictionary,
+          selecting the desired information. One of :
+
+          'index'
+            index in the standard AT coordinate vector
+          'code'
+            string representation
+          'label'
+            label for plot annotation
+          'unit'
+            coordinate unit
+          :py:obj:`None`
+            entire description dictionary
+
+    Returns:
+        descr : value of tuple[values]
+
+    Examples:
+
+        >>> axis_(('x','dp'), key='index')
+        (0, 4)
+
+        returns the indices in the standard coordinate vector
+
+        >>> dplabel = axis_('dp', key='label')
+        >>> print(dplabel)
+        $\delta$
+
+        returns the coordinate label for plot annotation
+
+        >>> axis_((0,'dp'))
+        ({'plane': 0, 'label': 'x', 'unit': ' [m]', 'code': 'x'},
+         {'plane': 4, 'label': '$\\delta$', 'unit': '', 'code': 'dp'})
+
+        returns the entire description directories
+
+    """
+    return _descr(_axis_def, axis, key=key)
+
+
+def plane_(plane: AxisDef, key: Optional[str] = None):
+    r"""Return plane descriptions
+
+    Parameters:
+        plane:          plane code or tuple[plane code] selecting planes.
+          planes for the 3 planes are:
+
+          0, 'h', 'H', 'x' for horizontal plane,
+
+          1, 'v', 'V', 'y' for vertical plane,
+
+          2, 'l', 'L' for the longitudinal plane
```

Comment:
Again why not `z` or `s` ? This is much more natural that `L` that is never used as it generally defines a length

Reply:
I initially wanted to distinguish axes  (x, p<sub>x</sub>,...) and planes (h <=> (x, p<sub>x</sub>)...). But you are right, it seems unavoidable to also use x, y, z for planes.